src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -0,0 +1,822 @@
+            {
+                try
+                {
+                    String id = matcher.group(1);
+                    TextChannel channel = getJDA().getTextChannelById(id);
+                    if (channel != null)
+                        channelMentions.add(channel);
+                }
+                catch (NumberFormatException ignored) {}
+            }
+            channelMentions = Collections.unmodifiableList(channelMentions);
+        }
+
+        return channelMentions;
+    }
+
+    @Override
+    public synchronized List<Role> getMentionedRoles()
+    {
+        if (roleMentions == null)
+        {
+            roleMentions = new ArrayList<>();
+            Matcher matcher = MentionType.ROLE.getPattern().matcher(content);
+            while (matcher.find())
+            {
+                try
+                {
+                    long id = MiscUtil.parseSnowflake(matcher.group(1));
+                    Role role = null;
+                    if (isFromType(ChannelType.TEXT)) // role lookup is faster if its in the same guild (no global map)
+                        role = getGuild().getRoleById(id);
+                    if (role == null)
+                        role = getJDA().getRoleById(id);
+                    if (role != null)
+                        roleMentions.add(role);
+                }
+                catch (NumberFormatException ignored) {}
+            }
+            roleMentions = Collections.unmodifiableList(roleMentions);
+        }
+
+        return roleMentions;
+    }
+
+    @Override
+    public List<Member> getMentionedMembers(Guild guild)
+    {
+        Checks.notNull(guild, "Guild");
+        List<User> mentionedUsers = getMentionedUsers();
+        List<Member> members = new ArrayList<>();
+        for (User user : mentionedUsers)
+        {
+            Member member = guild.getMember(user);
+            if (member != null)
+                members.add(member);
+        }
+
+        return Collections.unmodifiableList(members);
+    }
+
+    @Override
+    public List<Member> getMentionedMembers()
+    {
+        if (isFromType(ChannelType.TEXT))
+            return getMentionedMembers(getGuild());
+        else
+            throw new IllegalStateException("You must specify a Guild for Messages which are not sent from a TextChannel!");
+    }
+
+    @Override
+    public List<IMentionable> getMentions(MentionType... types)
+    {
+        if (types == null || types.length == 0)
+            return getMentions(MentionType.values());
+        List<IMentionable> mentions = new ArrayList<>();
+        // boolean duplicate checks
+        // not using Set because channel and role might have the same ID
+        boolean channel = false;
+        boolean role = false;
+        boolean user = false;
+        boolean emote = false;
+        for (MentionType type : types)
+        {
+            switch (type)
+            {
+                case EVERYONE:
+                case HERE:
```

Comment:
The returned list should also include `@here`/`@everyone` mentions.

Reply:
I disagree